src/ApiService/ApiService/OneFuzzTypes/Responses.cs

```diff
@@ -52,7 +52,9 @@ public record TaskSearchResult(
     List<NodeAssignment> Nodes,
     [property: JsonPropertyName("Timestamp")] // must retain capital T for backcompat
     DateTimeOffset? Timestamp
-) : BaseResponse();
+) : BaseResponse() {
+    public DateTimeOffset? StartTime => EndTime is DateTimeOffset endTime ? endTime.Subtract(TimeSpan.FromHours(Config.Task.Duration)) : null;
```

Comment:
In the issue, we say we want the start time to be when the task gets picked up by a node but this calculation doesn't cover for example a task `B` that depends on a prerequisite task `A`. Or, if the workset message stays in the pool queue while waiting for auto scale to kick in.

I think we'd get a more accurate picture by tracking StartTime around here:
https://github.com/microsoft/onefuzz/blob/6d8be1a83473c941565071d8f7988ea341e0b544/src/ApiService/ApiService/Functions/AgentCanSchedule.cs#L62-L72

We could store it as a column in the Tasks table.

Reply:
we set the `EndTime` at the when we receive the `OneStart` event. By making the `StartTime` dependent on the `EndTime` we capture the effective start